interfaces/builtin/repowerd.go

```diff
@@ -0,0 +1,335 @@
+dbus (send)
+  bus=system
+  path=/org/freedesktop/UPower{,/devices/**}
+  interface=org.freedesktop.DBus.Properties
+  member=Get{,All}
+  peer=(name=org.freedesktop.UPower),
+dbus (receive)
+  bus=system
+  path=/org/freedesktop/UPower{,/devices/**}
+  interface=org.freedesktop.DBus.Properties
+  member=PropertiesChanged
+  peer=(name=org.freedesktop.UPower),
+
+# Ofono
+dbus (send)
+  bus=system
+  path=/
+  interface=org.ofono.Manager
+  member=GetModems
+  peer=(name=org.ofono),
+dbus (send)
+  bus=system
+  interface=org.ofono.RadioSettings
+  member=SetProperty
+  peer=(name=org.ofono),
+dbus (send)
+  bus=system
+  path=/{,**}
+  interface=org.freedesktop.DBus.Properties
+  member=Get{,All}
+  peer=(name=org.ofono),
+dbus (receive)
+  bus=system
+  path=/{,**}
+  interface=org.freedesktop.DBus.Properties
+  member=PropertiesChanged
+  peer=(name=org.ofono),
+
+# Unity.Display
+dbus (send)
+  bus=system
+  interface=com.canonical.Unity.Display
+  path=/com/canonical/Unity/Display
+  member={TurnOn,TurnOff}
+  peer=(name=com.canonical.Unity.Display),
+
+# Unity.UserActivity
+dbus (receive)
+  bus=system
+  interface=com.canonical.Unity.UserActivity
+  path=/com/canonical/Unity/UserActivity
+  member=Activity,
+
+# Unity.PowerButton
+dbus (receive)
+  bus=system
+  interface=com.canonical.Unity.PowerButton
+  path=/com/canonical/Unity/PowerButton
+  member={Press,Release},
+dbus (send)
+  bus=system
+  interface=com.canonical.Unity.PowerButton
+  path=/com/canonical/Unity/PowerButton
+  member=LongPress,
+`)
+
+var repowerdConnectedPlugAppArmor = []byte(`
+# Description: Allow using repowerd service. Reserved because this gives
+#  privileged access to the repowerd service.
+# Usage: reserved
+
+#include <abstractions/dbus-strict>
+
+dbus (receive, send)
+  bus=system
+  path=/com/canonical/powerd
+  interface=org.freedesktop.DBus.{Properties,Introspectable},
+
+dbus (receive, send)
+  bus=system
+  path=/com/canonical/Unity/Screen
+  interface=org.freedesktop.DBus.{Properties,Introspectable},
+
+dbus (receive, send)
+  bus=system
+  path=/com/canonical/Unity/Screen
+  interface=com.canonical.Unity.Screen,
+
+dbus (receive, send)
+  bus=system
+  path=/com/canonical/powerd
+  interface=com.canonical.powerd,
+
+dbus (receive, send)
+  bus=system
+  path=/com/canonical/Unity/Screen
+  interface=com.canonical.Unity.Screen,
+`)
+
+var repowerdPermanentSlotSecComp = []byte(`
+# Description: Allow operating as the repowerd service. Reserved because this
+# gives
+#  privileged access to the system.
+# Usage: reserved
+
+# Can communicate with DBus system service
+sendto
+sendmsg
+recvfrom
+recvmsg
+`)
+
+var repowerdConnectedPlugSecComp = []byte(`
+# Description: Allow using repowerd service. Reserved because this gives
+#  privileged access to the repowerd service.
+# Usage: reserved
+
+# Can communicate with DBus system service
+sendmsg
+sendto
+recvfrom
+recvmsg
+`)
+
+var repowerdPermanentSlotDBus = []byte(`
+<policy user="root">
```

Comment:
This this policy directly from upstream or was modified? Please add a note like

<!-- DBus policy for upower (upstream version 0.99.4) -->